SCCM-TSEnvironment.psm1

```diff
@@ -440,19 +445,24 @@ function Show-TSErrorDialog()
         [Parameter(Mandatory=$true)]
         [long] $TimeoutInSeconds,
         [Parameter(Mandatory=$true)]
-        [bool] $ForceReboot
+        [bool] $ForceReboot,
+        [Parameter()] #Required for SCCM 1901 Tech Preview and newer clients
+        [string] $TSStepName
     )
 
     Confirm-TSProgressUISetup
     Confirm-TSEnvironmentSetup
 
-    if ($ForceReboot)
-    {
-        $Script:TaskSequenceProgressUi.ShowErrorDialog($OrganizationName, $Script:TaskSequenceEnvironment.Value("_SMSTSPackageName"), $CustomTitle, $ErrorMessage, $ErrorCode, $TimeoutInSeconds, 1)
+    [int]$Reboot = Switch($ForceReboot) {
+        $True {1}
+        $False {0}
     }
-    else
-    {
-        $Script:TaskSequenceProgressUi.ShowErrorDialog($OrganizationName, $Script:TaskSequenceEnvironment.Value("_SMSTSPackageName"), $CustomTitle, $ErrorMessage, $ErrorCode, $TimeoutInSeconds, 0)
+
+    If([string]::IsNullOrEmpty($TSStepName)) {
+        $Script:TaskSequenceProgressUi.ShowErrorDialog($OrganizationName, $Script:TaskSequenceEnvironment.Value("_SMSTSPackageName"), $CustomTitle, $ErrorMessage, $ErrorCode, $TimeoutInSeconds, $Reboot)
+    }
+    Else {
+        $Script:TaskSequenceProgressUi.ShowErrorDialog($OrganizationName, $Script:TaskSequenceEnvironment.Value("_SMSTSPackageName"), $CustomTitle, $ErrorMessage, $ErrorCode, $TimeoutInSeconds, $Reboot, $TSStepName)
```

Comment:
Would you be able to provide a visual of "where" this value shows up in the dialog? If I remember there were a few things that (while requiring a parameter) they didn't really "do anything".